### PR TITLE
[world-vercel] Use stream close control frame to decide whether to reconnect to stream

### PR DESCRIPTION
**Pull request by VaguelySerious**

### CODE REVIEW

File: packages/world-vercel/src/utils.ts [hunk 1]
```diff
@@ -61,7 +61,8 @@ function httpLog(
  *
  * Example: 'https://workflow-server-git-branch-name.vercel.sh'
  */
-const WORKFLOW_SERVER_URL_OVERRIDE = '';
+const WORKFLOW_SERVER_URL_OVERRIDE =
+  'https://workflow-server-git-peter-stream-control-frame.vercel.sh';
```
Comment (TooTallNate) [inline packages/world-vercel/src/utils.ts]: Don't forget to remove this before merging <picture data-single-emoji=":plz:" title=":plz:">:plz:</picture>

Comment (TooTallNate) [inline packages/world-vercel/src/utils.ts]: **Blocking**: This hardcodes `WORKFLOW_SERVER_URL_OVERRIDE` to a preview deployment URL. This must be reverted to `''` before merging — it would route all production traffic to a preview branch deployment.

(I left a comment about this earlier, flagging again as blocking since it's still present.)

File: packages/world-vercel/src/streamer.ts [hunk 2]
```diff
@@ -182,21 +254,105 @@ export function createStreamer(config?: APIConfig): Streamer {
...
+              try {
+                result = await reader.read();
+              } catch {
+                // Network error — not a clean close. Forward any buffered data
+                // and propagate the error so consumers can handle it.
+                if (tailBuffer.length > 0) {
+                  controller.enqueue(tailBuffer);
+                  tailBuffer = new Uint8Array(0);
+                }
+                controller.close();
```
Comment (TooTallNate) [inline packages/world-vercel/src/streamer.ts]: Non-blocking concern: On network error, the code swallows the error and calls `controller.close()` instead of `controller.error(err)`. This means consumers of the returned ReadableStream see a clean close rather than an error, and have no way to know the stream was truncated.

The reconnect test asserts the partial data is received and only 1 fetch is made, which is correct for "no retry on errors." But silently closing on error could be a footgun — the consumer (the workflow VM) would think it received the complete stream.

Consider either:
- Propagating the error via `controller.error(err)` so consumers can handle it
- Or at minimum, logging a warning

File: packages/world-vercel/src/streamer.ts [hunk 3]
```diff
@@ -182,21 +254,105 @@ export function createStreamer(config?: APIConfig): Streamer {
...
 
       async get(runId: string, name: string, startIndex?: number) {
-        const httpConfig = await getHttpConfig(config);
-        const url = getStreamUrl(name, runId, httpConfig);
-        if (typeof startIndex === 'number') {
-          url.searchParams.set('startIndex', String(startIndex));
-        }
-        const response = await fetch(url, {
-          headers: httpConfig.headers,
+        let currentStartIndex = startIndex ?? 0;
```
Comment (TooTallNate) [inline packages/world-vercel/src/streamer.ts:257]: Non-blocking: There's no upper bound on reconnection attempts. If the server keeps returning `done=false` (e.g., due to a bug where the stream never completes), this will loop indefinitely. Consider adding a `maxReconnects` counter (e.g., 50 — enough for ~100 minutes of streaming at 2-minute timeouts) with a warning log when exhausted.

**Review by TooTallNate — CHANGES_REQUESTED** [review]
The reconnection design is well thought out. The hold-back buffer approach for control frame detection is correct and handles all edge cases I traced through (split frames, coalesced chunks, empty streams, short data). The test coverage is thorough. But there's one blocking issue and a couple of concerns.

**Cross-reference with server-side PR 367**: I reviewed the server PR and left a `REQUEST_CHANGES` there too. The server currently only emits the control frame on timeout (`enableControlFrame && maxDurationReached`), NOT on natural stream completion. This client correctly handles that via the "no control frame" fallback path — so the two PRs are compatible as-is. However, the server PR has 6 failing integration tests due to a test/code mismatch introduced in its last commits. This client PR should not be merged until the server PR is resolved and deployed.

**Changes reviewed:**
1. **`parseStreamControlFrame`** (`streamer.ts`): Correct mirror of server's format. Checks zero-marker + magic footer, parses flags and nextIndex.
2. **Hold-back buffer** (`streamer.ts:278-348`): Retains last 13 bytes during streaming to detect the control frame at close. Handles split frames, coalesced chunks, and backward compat (no frame) gracefully.
3. **Reconnection loop** (`streamer.ts:335-345`): On `done=false`, reconnects from `control.nextIndex`. Clean loop.
4. **Error handling** (`streamer.ts:289-296`): Network errors flush the buffer and close without retrying. Correct.
5. **Unit tests** (`streamer.test.ts`): Good coverage of `parseStreamControlFrame` edge cases.
6. **Integration tests** (`streamer-reconnect.test.ts`): Comprehensive — covers single reconnect, multiple reconnects, split frames, coalesced frames, empty streams, backwards compat, network errors, and non-200 responses.

File: packages/world-vercel/src/streamer.ts (hunk 2 above)
Comment (VaguelySerious) [reply packages/world-vercel/src/streamer.ts]: Fixed in 6ee6e7de97 — network errors now propagate via `controller.error(err)` instead of silently closing.

File: packages/world-vercel/src/streamer.ts (hunk 3 above)
Comment (VaguelySerious) [reply packages/world-vercel/src/streamer.ts:257]: Fixed in 6ee6e7de97 — added `MAX_RECONNECTS = 50` cap (~100 min at 2-min server timeout). Errors with a descriptive message when exhausted.

**Review by TooTallNate — APPROVED** [review]
All feedback addressed. LGTM.

**What changed since my last review:**
1. **`WORKFLOW_SERVER_URL_OVERRIDE`** — Reverted. `utils.ts` is no longer in the diff.
2. **Error propagation** — `controller.close()` changed to `controller.error(err)` on network errors. Test updated to assert `rejects.toThrow('connection reset')`.
3. **Reconnection cap** — Added `MAX_RECONNECTS = 50` (~100 min at 2-min timeouts). Errors with a clear message when exhausted.

All CI checks pass (unit tests on both ubuntu and windows, build, E2E).
